src/coreclr/vm/peassembly.cpp

```diff
@@ -320,12 +320,36 @@ void PEAssembly::ConvertMDInternalToReadWrite()
     }
     CONTRACTL_END;
 
-    IMDInternalImport *pOld;            // Old (current RO) value of internal import.
-    IMDInternalImport *pNew = NULL;     // New (RW) value of internal import.
-
-    // Take a local copy of *ppImport.  This may be a pointer to an RO
+    // Take a local copy of the assembly's MD import.  This may be a pointer to an RO
     //  or to an RW MDInternalXX.
-    pOld = m_pMDImport;
+    IMDInternalImport *pNew = NULL;                                 // New (RW) value of internal import.
+    IMDInternalImport *pOld = m_pMDImport;                          // Old (current RO) value of internal import.
+    IMDInternalImport *pConvertedImport = m_pConvertedMDImport;     // Pointer to a RW interna metadata that might not be actively in use.
+
+
+    if (pConvertedImport != NULL)
+    {
+        if (swapForRWMDImport && pOld != pConvertedImport)
+        {
+            // We had converted the metadata before, now we are just requested
+            // to replace the RO view we had with the converted one.
+            if (InterlockedCompareExchangeT(&m_pMDImport, pConvertedImport, pOld) == pOld)
+            {
+                m_pMDImport->AddRef();
```

Comment:
Why do we need another `AddRef()`? This was previously a member, right? Who is going to release this?

Reply:
There are 3 scenarios:

- `m_pMDImport` is RO, `m_pConvertedMDImport` NULL. Releasing m_pMDImport in the destructor is enough.
- `m_pMDImport` is RO, `m_pConvertedMDImport` is RW. Both get released in the destructor separately. 
- `m_pMDImport` is RW, and `m_pConvertedMDImport` points to the same. 

The AddRef I though was necessary since now `m_pConvertedMDImport == m_pMDImport`, and the destructor will release both, so I need a refcount of 2 to keep the RO saved as "user context data" alive until both are gone.

Reply:
If this AddRef is logically part of `SetUserContextData(pOld)`, it should be `pOld->AddRef()` (same argument as what is passed into `SetUserContextData`). Comment would be nice too.

Reply:
I've added a clarifying comment for this one.